src/test/rbf_tests.cpp

```diff
@@ -0,0 +1,211 @@
+    // EntriesAndTxidsDisjoint uses txids, not wtxids.
+    BOOST_CHECK(EntriesAndTxidsDisjoint({entry2}, {tx2->GetWitnessHash()}, unused_txid) == std::nullopt);
+    // If entry2 is an ancestor of a tx, that tx cannot replace entry1.  However,
+    // EntriesAndTxidsDisjoint uses the ancestors directly. It does not calculate descendants.
+    BOOST_CHECK(EntriesAndTxidsDisjoint(set_12_normal, {tx1->GetHash()}, unused_txid).has_value());
+    BOOST_CHECK(EntriesAndTxidsDisjoint(set_12_normal, {tx2->GetHash()}, unused_txid).has_value());
+    BOOST_CHECK(EntriesAndTxidsDisjoint({entry2}, {tx1->GetHash()}, unused_txid) == std::nullopt);
+
+    // Tests for PaysForRBF
+    const auto incremental_relay_feerate{CFeeRate(DEFAULT_INCREMENTAL_RELAY_FEE)};
+    const auto higher_relay_feerate{CFeeRate(2, 1)};
+    // Must pay at least as much as the original.
+    BOOST_CHECK(PaysForRBF(/*original_fees=*/high_fee,
+                           /*replacement_fees=*/high_fee,
+                           /*replacement_vsize=*/1,
+                           /*relay_fee=*/CFeeRate(0),
+                           /*txid=*/unused_txid)
+                           == std::nullopt);
+    BOOST_CHECK(PaysForRBF(high_fee, high_fee - 1, 1, CFeeRate(0), unused_txid).has_value());
+    BOOST_CHECK(PaysForRBF(high_fee + 1, high_fee, 1, CFeeRate(0), unused_txid).has_value());
+    // Additional fees must cover the replacement's vsize at incremental relay fee
+    BOOST_CHECK(PaysForRBF(high_fee, high_fee + 1, 2, incremental_relay_feerate, unused_txid).has_value());
+    BOOST_CHECK(PaysForRBF(high_fee, high_fee + 2, 2, incremental_relay_feerate, unused_txid) == std::nullopt);
+    BOOST_CHECK(PaysForRBF(high_fee, high_fee + 2, 2, higher_relay_feerate, unused_txid).has_value());
+    BOOST_CHECK(PaysForRBF(high_fee, high_fee + 4, 2, higher_relay_feerate, unused_txid) == std::nullopt);
+    BOOST_CHECK(PaysForRBF(low_fee, high_fee, 99999999, incremental_relay_feerate, unused_txid).has_value());
+
+    // Tests for GetEntriesForConflicts
+    CTxMemPool::setEntries all_parents{entry1, entry3, entry5, entry7, entry8};
+    CTxMemPool::setEntries all_children{entry2, entry4, entry6};
+    std::vector<CTransactionRef> parent_inputs({m_coinbase_txns[0], m_coinbase_txns[1], m_coinbase_txns[2],
+                                                m_coinbase_txns[3], m_coinbase_txns[4]});
+    CTransactionRef conflicts_with_parents = make_tx({50 * CENT}, parent_inputs);
+    CTxMemPool::setEntries all_conflicts;
+    BOOST_CHECK(GetEntriesForConflicts(/*tx=*/ *conflicts_with_parents.get(),
+                                       /*pool=*/ pool,
+                                       /*iters_conflicting=*/ all_parents,
+                                       /*all_conflicts=*/ all_conflicts) == std::nullopt);
+    BOOST_CHECK(all_conflicts == all_entries);
+    all_conflicts.clear();
+
+    add_descendants(tx2, pool, 23);
```

Comment:
QoL improvement for future readers/editors would be accumulating these constants

Reply:
Done